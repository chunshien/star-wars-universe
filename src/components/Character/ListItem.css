.container {
    cursor: pointer;
    width: 320px;
    float: left;
    padding: 20px;
    margin: 10px;
    border-radius: 5px;
    box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
}

.item {
    display: table;
}

.section {
    display: table-cell;
    vertical-align: top;
    padding: 10px;
}

.avatar {
    width: 50px;
    height: 50px;
    padding: 10px;
    border-radius: 40px;
    border-color: black;
    box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
    object-fit: contain;
}