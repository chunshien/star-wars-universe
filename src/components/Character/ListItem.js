import React, {useCallback} from 'react';
import styled from 'styled-components'
import avatar from '../../assets/avatar.png';

const Container = styled.div`
  cursor: pointer;
  width: 48%;
  background-color: white;
  float: left;
  box-sizing: border-box;
  padding: 20px;
  margin: 1%;
  border-radius: 5px;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
`;

const Item = styled.div`
  display: table;
`;

const Section = styled.div`
  display: table-cell;
  vertical-align: top;
  padding: 10px;
`;

const Avatar = styled.img`
  width: 50px;
  height: 50px;
  padding: 10px;
  border-radius: 40px;
  border-color: black;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
  object-fit: contain;
`;

function ListItem(props) {
  const getColor = (skinType) => {
    if (skinType.indexOf('fair') >= 0) {
      return '#ffe0bd';
    } 
    else if (skinType.indexOf('pale') >= 0) {
      return '#fdf5e2';
    }
    else if (skinType.indexOf('light') >= 0) {
      return '#ffd6a4';
    }
    else if (skinType.indexOf('green') >= 0) {
      return 'green';
    }
    else if (skinType.indexOf('brown') >= 0) {
      return 'brown';
    }
    else {
      return skinType;
    }
  }

  const handleClick = useCallback(() => {
    props.onClick(props.data.url);
  }, [props])

  if(props.data){
    return (
      <Container onClick={handleClick}>
          <Item>
            <Section>
              <Avatar className={'avatar'} 
                style={{backgroundColor: getColor(props.data.skin_color)}} 
                src={avatar} alt={props.data.name} />
            </Section>
            <Section>
              <div>Name: {props.data.name}</div>
              <div>Height: {props.data.height}</div>
              <div>Gender: {props.data.gender}</div>
              <div>Mass: {props.data.mass}</div>
            </Section>
          </Item>        
      </Container>
    );
  }
  else {
    return <div></div>
  }
}

export default ListItem;
