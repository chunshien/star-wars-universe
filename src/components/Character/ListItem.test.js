import React from 'react';
import ListItem from './ListItem';
import renderer from 'react-test-renderer';
const testData = require('../../assets/test-data/peoples')

describe('Component - ListItem Test', () => {
    it('renders ListItem without data without crashing', () => {
        const component = renderer.create(
            <ListItem/>    
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot() 
    })
    it('renders ListItem with data without crashing', () => {    
        const component = renderer.create(
            <ListItem
                data={testData.results[0]}
            />    
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot() 
    })
})

