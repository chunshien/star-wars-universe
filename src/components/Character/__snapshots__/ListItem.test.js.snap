// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ListItem Test renders ListItem with data without crashing 1`] = `
<div
  className="sc-AxjAm iFUVjd"
  onClick={[Function]}
>
  <div
    className="sc-AxirZ ehqiCf"
  >
    <div
      className="sc-AxiKw iPnqiR"
    >
      <img
        alt="Luke Skywalker"
        className="sc-AxhCb jvfHjD avatar"
        src="avatar.png"
        style={
          Object {
            "backgroundColor": "#ffe0bd",
          }
        }
      />
    </div>
    <div
      className="sc-AxiKw iPnqiR"
    >
      <div>
        Name: 
        Luke Skywalker
      </div>
      <div>
        Height: 
        172
      </div>
      <div>
        Gender: 
        male
      </div>
      <div>
        Mass: 
        77
      </div>
    </div>
  </div>
</div>
`;

exports[`Component - ListItem Test renders ListItem without data without crashing 1`] = `<div />`;
